packages/build/src/report/statsd.ts

```diff
@@ -1,40 +1,30 @@
 import { promisify } from 'util'
 
 import slugify from '@sindresorhus/slugify'
-import StatsdClient from 'statsd-client'
+import { StatsD, Tags } from 'hot-shots'
 
-// TODO: replace with `timers/promises` after dropping Node < 15.0.0
-const pSetTimeout = promisify(setTimeout)
-
-// See https://github.com/msiebuhr/node-statsd-client/blob/45a93ee4c94ca72f244a40b06cb542d4bd7c3766/lib/EphemeralSocket.js#L81
-const CLOSE_TIMEOUT = 11
-
-// The socket creation is delayed until the first packet is sent. In our
-// case, this happens just before `client.close()` is called, which is too
-// late and make it not send anything. We need to manually create it using
-// the internal API.
-export const startClient = async function (host: string, port: number): Promise<StatsdClient> {
-  const client = new StatsdClient({ host, port, socketTimeout: 0 })
-
-  // @ts-expect-error using internals :D
-  await promisify(client._socket._createSocket.bind(client._socket))()
```

Comment:
Because `socketTimeout` was set to 0 we never actually send the values as this value also is set as send-interval:
> **socketTimeout** Dual-use timer. Will flush metrics every interval. For UDP, it auto-closes the socket after this long without activity (default 1000 ms; 0 disables this).

With `hot-shot` the interval is now `1000ms` so if there are metrics to send it will wait for 1s for more metrics before sending them.

And as we await the close now, we make sure that everything is sent when closing the socket.